src/utils/hooks/useAutoLogout.js

```diff
@@ -0,0 +1,30 @@
+import { getLogoutUrl } from 'components/auth/provider/component';
+import { decodeJwt } from 'oidc-spa';
+import { OIDC } from 'utils/constants';
+import { listenActivity } from 'utils/events';
+import { environment } from '../read-env-vars';
+
+const { AUTH_TYPE } = environment;
+
+export const useAutoLogout = ({ oidc }) => {
+  const { isUserLoggedIn, logout } = oidc;
+  if (isUserLoggedIn && AUTH_TYPE === OIDC) {
+    (async function callee() {
+      const { exp: expirationTime } = decodeJwt(oidc?.getTokens()?.accessToken);
+      const logoutTimeInMiliseconds = expirationTime * 1000 - Date.now();
+
+      const logoutTimeout = setTimeout(async () => {
+        console.log(
+          `No user activity detected during ${
+            logoutTimeInMiliseconds / 1000
+          } seconds. Logout user.`
+        );
+        await logout({ redirectTo: 'specific url', url: getLogoutUrl() });
+      }, logoutTimeInMiliseconds);
+
+      await listenActivity();
+      clearTimeout(logoutTimeout);
+      callee();
+    })();
+  }
+};
```

Comment:
I try to do this, but the useEffect is running only once: ok.
So you don't listen activity anymore, so you user is logout after timeout even if he is active.

I try do add `timerId` to dependency array but it's worst, it causes Loop : `Maximum update depth exceeded. This can happen when a component calls setState inside useEffect`

Reply:
Ok i think we should do [this](https://github.com/keycloakify/oidc-spa/issues/14#issuecomment-1923521386) unless [library handle case ](https://github.com/keycloakify/oidc-spa/issues/17)

Reply:
I implemented this in the new version